libgc/mark_rts.c

```diff
@@ -369,6 +369,9 @@ ptr_t p;
 
 ptr_t GC_approx_sp()
 {
+#ifdef __GNUC__
```

Comment:
AFAIU the issue is caused by a change in how gcc emits code - it may be that the behavior didn't change / changed differently for other OS/platform (I only ran into it because my local install got upgraded to gcc5), but this primitive should be supported everywhere gcc gccs.